src/bidiMapper/domains/script/scriptEvaluator.ts

```diff
@@ -251,154 +397,129 @@ export class ScriptEvaluator {
         return {objectId: argEvalResult.result.objectId};
       }
 
-      // TODO(#375): dispose of nested objects.
+      case 'channel': {
+        const channelName = argumentValue.value;
 
-      default:
-        throw new Error(
-          `Value ${JSON.stringify(argumentValue)} is not deserializable.`
+        const argEvalResult = await realm.cdpClient.sendCommand(
+          'Runtime.callFunctionOn',
+          {
+            functionDeclaration: String(() => {
+              const queue: string[] = [];
+              let resolver: (message: string | PromiseLike<string>) => void;
+
+              return {
+                /**
+                 * Gets a promise, which is resolved as soon as a message occurs
+                 * in the queue.
+                 */
+                async getMessage(): Promise<string> {
+                  if (queue.length > 0) {
+                    return queue.shift()!;
+                  }
+                  const promise = new Promise<string>((resolve) => {
+                    resolver = resolve;
+                  });
+                  promise.then(() => queue.shift());
+                  return promise;
+                },
+
+                /**
+                 * Adds a message to the queue.
+                 * If the queue is initially empty, resolve the pending promise.
+                 */
+                sendMessage(message: string) {
+                  queue.push(message);
+                  if (queue.length === 1) {
+                    resolver(message);
+                  }
+                },
+              };
+            }),
+            returnByValue: false,
+            executionContextId: realm.executionContextId,
+            generateWebDriverValue: false,
+          }
         );
-    }
-  }
-
-  /**
-   * Serializes a given CDP object into BiDi, keeping references in the
-   * target's `globalThis`.
-   * @param cdpRemoteObject CDP remote object to be serialized.
-   * @param resultOwnership Indicates desired ResultOwnership.
-   * @param realm
-   */
-  async serializeCdpObject(
-    cdpRemoteObject: Protocol.Runtime.RemoteObject,
-    resultOwnership: Script.ResultOwnership,
-    realm: Realm
-  ): Promise<CommonDataTypes.RemoteValue> {
-    const arg = cdpRemoteObjectToCallArgument(cdpRemoteObject);
-
-    const cdpWebDriverValue: Protocol.Runtime.CallFunctionOnResponse =
-      await realm.cdpClient.sendCommand('Runtime.callFunctionOn', {
-        functionDeclaration: String((obj: unknown) => obj),
-        awaitPromise: false,
-        arguments: [arg],
-        generateWebDriverValue: true,
-        executionContextId: realm.executionContextId,
-      });
-    return realm.cdpToBidiValue(cdpWebDriverValue, resultOwnership);
-  }
+        const channelHandle = argEvalResult.result.objectId;
+
+        // Long-poll the message queue asynchronously.
+        (async () => {
+          for (;;) {
+            const message = await realm.cdpClient.sendCommand(
```

Comment:
Please add TODO to remove this loop after the realm is destroyed. You can rely on the CDP throwing exception in such a case.